src/pfe/portal/modules/MetricsService.js

```diff
@@ -122,19 +122,33 @@ async function injectMetricsCollectorIntoApplicationJava(pathToApplicationJava)
 
 function getNewContentsOfApplicationJava(originalContents) {
   const splitOriginalContents = originalContents.split('\n');
+  const componentScanPresent = splitOriginalContents.findIndex(line => {
+    return (line.indexOf('@ComponentScan(basePackages = {') != -1)
+  });
+  const metricsCollectorPresent = splitOriginalContents.some(line => {
+    return (line.indexOf('@ComponentScan(basePackages = {') != -1) && (line.indexOf('com.ibm.javametrics.codewind.spring') != -1)
+  });
+
+  if (metricsCollectorPresent) {
+    return originalContents;
+  }
 
-  const packageString = splitOriginalContents.find(line => line.includes('package'));
-  const packageName = packageString // 'package NAME'
-    .split(' ')[1] // 'NAME;'
-    .slice(0, -1); // 'NAME'
-
-  const indexOfSpringBootApplication = splitOriginalContents.findIndex(line => line === springAppAnnotation);
-  const metricsCollectorAnnotation = `@ComponentScan(basePackages = {"${packageName}", "com.ibm.javametrics.spring"})`;
-
+  // metrics collector not present, so...
   let newApplicationJava = deepClone(splitOriginalContents);
-  newApplicationJava.splice(indexOfSpringBootApplication + 1, 0, metricsCollectorAnnotation);
+  if (componentScanPresent != -1) {
+    // alter what's already in there
+    newApplicationJava[componentScanPresent] = newApplicationJava[componentScanPresent].replace('{', '{"com.ibm.javametrics.codewind.spring", ');
```

Comment:
as `componentScanPresent` isn't a boolean nor used like one, should it have a different name?